examples/spec/integration/query_workflow_spec.rb

```diff
@@ -0,0 +1,18 @@
+require 'workflows/query_workflow'
+
+describe QueryWorkflow, :integration do
+  subject { described_class }
+
+  it 'returns the correct result for the queries' do
```

Comment:
Test currently only covers the named query "state." Can you add a test to confirm the default `on_query` handler runs? Lastly, add a test to confirm that two named `on_query` handlers can live side by side and be invoked separately?

Time permitting, I will add these tests or we can do a followup.

Reply:
BTW, I know the default on_query handler is implicitly tested here via the cancel behavior. But it isn't clear at first glance. I prefer explicit tests.

Reply:
I expanded the integration test scenarios a bit in the latest round of changes